catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -390,6 +395,69 @@ public void testExecuteQueryCacheDisabled() throws Exception {
                 .size(), is(0));
     }
 
+    @Test
+    public void testMaxQueryPaging() throws Exception {
```

Comment:
So I guess my point was that we're trying to test all the different paths for the `while` loop here: https://github.com/codice/ddf/pull/2195/files#diff-675e304e397476c2b776d7b512286af6R293.

Here's what I had in mind, but there could be more:
1. The first query returns everything requested, e.g., page size = 100, total hits = 100 and result size = 100. In other words, we don't even go inside the while loop.
2. We need to query the Catalog multiple times until we get to the page size requested, which is less than the total hits, e.g., page size = 100, total hits = 300 and result sizes = 50, 50. In this case, we go in the while loop once and leave because `intialResultsSize < request.getQuery().getPageSize()`
3. The test you currently have where we need to query the Catalog until we hit the total hits of 1,000, which covers the case where we exit the while loop because `intialResultsSize < response.getHits()`

We may also want to have some tests where the start index for the query is not 1 with the above variations to make sure we stop when expected. In other words, to test that lines https://github.com/codice/ddf/pull/2195/files#diff-675e304e397476c2b776d7b512286af6R295 and https://github.com/codice/ddf/pull/2195/files#diff-675e304e397476c2b776d7b512286af6R298 work as expected.

Hope this makes sense.